ddsketch-ffi/src/error.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use ddcommon_ffi::CString;
+use std::fmt::Display;
+
+/// Represent error codes that `DDSketchError` struct can hold
+#[repr(C)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DDSketchErrorCode {
+    InvalidArgument,
+    InvalidInput,
+    Internal,
+}
+
+impl Display for DDSketchErrorCode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::InvalidArgument => write!(f, "Invalid argument provided"),
+            Self::InvalidInput => write!(f, "Invalid input"),
+            Self::Internal => write!(f, "Internal error"),
+        }
+    }
+}
+
+/// Structure that contains error information that DDSketch FFI API can return.
+#[repr(C)]
+#[derive(Debug)]
+pub struct DDSketchError {
+    pub code: DDSketchErrorCode,
+    pub msg: CString,
+}
```

Comment:
If we do keep `DDSketchError` (see my note below), I believe it should be renamed `ddog_ddsketch_DDSketchError` / `ddog_ddsketch_DDSketchErrorCode`, right now it's missing the prefixes in the `.h` files :eyes:

Reply:
I got rid of DDSketchError for now.